### DISCUSSION TITLE
All categories & Max pages

### DISCUSSION THREAD

**Discussion by adamel339944** [aapatre/Automatic-Udemy-Course-Enroller-GET-PAID-UDEMY-COURSES-for-FREE#363] [Q&A]
Hello, at the end of running the script I got this:

Unwanted Category:          70

Can you please list all categories or is there any setting to ignore this and just enroll on all courses? I've got all the big categories but I guess this is not enough?

I mean I though if I put in the categories for example: DEVELOPMENT, I will get all the couses but I need to also get the sub-cattegories in?

Also how do I get all the pages?

I've tried to run the commends like: 

python3 udemy_enroller.py --max-pages 500
python3 udemy_enroller.py --max-pages500
python3 udemy_enroller.py --max-pages=300

And I get this error:

PS C:\Users\adam\Desktop\Automatic-Udemy-Course-Enroller-GET-PAID-UDEMY-COURSES-for-FREE-master (1)\Automatic-Udemy-Course-Enroller-GET-PAID-UDEMY-COURSES-for-FREE-master> python3 udemy_enroller.py --max-pages 500
usage: udemy_enroller.py [-h] --browser {chromium,internet_explorer,firefox,ff,google-chrome,chrome,opera,ie,edge} [--freebiesglobal] [--tutorialbar] [--discudemy] [--coursevania] [--max-pages MAX_PAGES] [--delete-settings] [--delete-cookie] [--debug]
udemy_enroller.py: error: the following arguments are required: --browser
PS C:\Users\adam\Desktop\Automatic-Udemy-Course-Enroller-GET-PAID-UDEMY-COURSES-for-FREE-master (1)\Automatic-Udemy-Course-Enroller-GET-PAID-UDEMY-COURSES-for-FREE-master> python3 udemy_enroller.py --max-pages500
usage: udemy_enroller.py [-h] --browser {chrome,ff,internet_explorer,google-chrome,firefox,ie,chromium,edge,opera} [--freebiesglobal] [--tutorialbar] [--discudemy] [--coursevania] [--max-pages MAX_PAGES] [--delete-settings] [--delete-cookie] [--debug]
udemy_enroller.py: error: the following arguments are required: --browser
PS C:\Users\adam\Desktop\Automatic-Udemy-Course-Enroller-GET-PAID-UDEMY-COURSES-for-FREE-master (1)\Automatic-Udemy-Course-Enroller-GET-PAID-UDEMY-COURSES-for-FREE-master> python3 udemy_enroller.py --max-pages=300
usage: udemy_enroller.py [-h] --browser {internet_explorer,ie,chrome,opera,chromium,firefox,edge,google-chrome,ff} [--freebiesglobal] [--tutorialbar] [--discudemy] [--coursevania] [--max-pages MAX_PAGES] [--delete-settings] [--delete-cookie] [--debug]
udemy_enroller.py: error: the following arguments are required: --browser

**Comment by cullzie** [comment]
Hey @xxDonDraper ,

The settings are saved under your home directory in a folder named  `.udemy_enroller` and a file named `settings.yaml`
If you can't find it there is an option to delete the settings file `--delete-settings` and you will be prompted to regenerate it with the categories you want. If you leave categories empty it will enroll to all.

There is no setting to go through all pages. Once you go beyond page 5 there are generally very few coupons that aren't expired.
`python3 udemy_enroller.py --browser=chrome --max-pages=500`

As for your browser issue the script must be run with a browser since v4 as there were changes on Udemy side.
To pass in a browser `--browser=chrome` (at the moment only chrome based browsers are working unfortunately)
